### ISSUE TITLE
Bug: serial deployment fails without failed tasks

### ISSUE THREAD

**Issue by arminreh** [ansible/ansible#7232]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.5.3
##### Environment:

Deployment from OSX 10.9.2
Ansible target is Freebsd 9.2
##### Summary:

When I run a serial task for a rolling update it stops with "FATAL: all hosts have already failed -- aborting" as soon as the first task returns a "ok:".

When running the task limited on one host after the other it works fine, when running it on two or more at the same time it breaks after gathering facts and returning from the first task on one system.
##### Steps To Reproduce:

The call to the playbook:
    ansible-playbook -i ansible/accelerated -l tracker-three-ams.acc,tracker-four-ams.acc ansible/tracker.yml

The tracker.yml definition:

```

---
- hosts: tracker
  serial: 1
  max_fail_percentage: 0
  remote_user: deploy
  sudo: yes
  roles:
     - common
     - tracker
     - monit
```
##### Expected Results:

```
ansible-playbook -i ansible/accelerated -l tracker-three-ams.acc,tracker-four-ams.acc ansible/tracker.yml

PLAY [tracker] ****************************************************************

GATHERING FACTS ***************************************************************
ok: [tracker-four-ams.acc]
ok: [tracker-three-ams.acc]

TASK: [common | install hosts file] *******************************************
ok: [tracker-four-ams.acc]
ok: [tracker-three-ams.acc]
```
##### Actual Results:

```
ansible-playbook -i ansible/accelerated -l tracker-three-ams.acc,tracker-four-ams.acc ansible/tracker.yml

PLAY [tracker] ****************************************************************

GATHERING FACTS ***************************************************************
ok: [tracker-four-ams.acc]
ok: [tracker-three-ams.acc]

TASK: [common | install hosts file] *******************************************
ok: [tracker-four-ams.acc]

FATAL: all hosts have already failed -- aborting

PLAY RECAP ********************************************************************
tracker-four-ams.acc       : ok=2    changed=0    unreachable=0    failed=0
tracker-three-ams.acc      : ok=1    changed=0    unreachable=0    failed=0
```

Thank you for any help!

**Comment by jimi-c** [comment]
You've set the max_fail_percentage to 0, which tells ansible to fail as soon as that percentage is exceeded. Since it is 0, it will always be exceeded and thus ansible will always fail. This is a common issue when using `serial: 1`, since you're telling ansible to operate on just one host at a time. In this case, the max_fail_percentage will always be exceeded if any system fails (since 1 out of 1 is 100%), but for this use case it seems that is what you want?

**Comment by arminreh** [comment]
Thanks for the fast reply.
In the end I want to stop the task as soon as the first hosts fails, so the other hosts can't be broken as well.
It turns out max_fail_percentage: 50 works when I have two hosts in the file, 49 does not.
But I fear this will not stop the playbook if one of the updates fails.
For three it has to be 66 or 67 to work.

Would be nice to have a solution that does not depend on the number of hosts in the list

**Comment by jimi-c** [comment]
You can continue to use `serial: 1`, just don't set the max_fail_percentage (it defaults to 100%). In that case, any failure will cause the play to stop. If you want it to only stop if the first host in the list works, you can split the play into two plays, one where you use `hosts: group[0]` and the second uses `hosts: group`. Assuming you're not doing anything that can't be re-run this will stop if the first play fails only operating on the first host, and then will update everything in parallel.

**Comment by jimi-c** [comment]
Based on the above, I'm going to go ahead and close this issue, since it is not a bug. Please let us know if you have any further questions. Thanks!
